python/cudf/cudf/core/dtypes.py

```diff
@@ -301,3 +302,29 @@ def _validate(cls, precision, scale=0):
             )
         if abs(scale) > precision:
             raise ValueError(f"scale={scale} exceeds precision={precision}")
+
+
+class IntervalDtype(StructDtype):
+    name = "interval"
```

Comment:
Should we have a `__repr__` that shows it as `interval[int64]` or `interval[int32]` for example? I imagine we don't intend to support cases where the left and right dtypes are different?